misc/metrics-extensions/modules/ballerina-prometheus-extension/src/main/java/org/ballerinalang/observe/metrics/extension/prometheus/PrometheusMeterRegistryProvider.java

```diff
@@ -58,7 +58,12 @@ public MeterRegistry get() {
         PrometheusMeterRegistry registry = new PrometheusMeterRegistry(new BallerinaPrometheusConfig());
         String hostname = configRegistry.getAsString(METRICS_HOSTNAME);
         boolean hostnameAvailable = hostname != null && !hostname.isEmpty();
-        int configuredPort = Math.toIntExact(configRegistry.getAsInt(METRICS_PORT));
+        int configuredPort;
+        try {
+            configuredPort = Integer.parseInt(configRegistry.getConfigOrDefault(METRICS_PORT, String.valueOf(0)));
+        } catch (IllegalArgumentException e) {
```

Comment:
This should be NumberFormatException https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#parseInt-java.lang.String-

Reply:
NumberFormatException extends from IllegalArgumentException